x-pack/libbeat/autodiscover/providers/aws/ec2/provider.go

```diff
@@ -0,0 +1,147 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package ec2
+
+import (
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/ec2iface"
+	"github.com/gofrs/uuid"
+	"github.com/pkg/errors"
+
+	awsauto "github.com/elastic/beats/x-pack/libbeat/autodiscover/providers/aws"
+	awscommon "github.com/elastic/beats/x-pack/libbeat/common/aws"
+
+	"github.com/elastic/beats/libbeat/autodiscover"
+	"github.com/elastic/beats/libbeat/autodiscover/template"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/common/bus"
+	"github.com/elastic/beats/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+func init() {
+	autodiscover.Registry.AddProvider("aws_ec2", AutodiscoverBuilder)
+}
+
+// Provider implements autodiscover provider for aws EC2s.
+type Provider struct {
+	config        *awsauto.Config
+	bus           bus.Bus
+	builders      autodiscover.Builders
+	appenders     autodiscover.Appenders
+	templates     *template.Mapper
```

Comment:
these can be removed

Reply:
Seems like templates is still needed?!

Reply:
yes! sorry :pray:

Reply:
ha no problem. I just want to make sure the right amount is removed 😬 Thanks!